src/Microsoft.OData.Core/JsonLight/ODataJsonLightErrorDeserializer.cs

```diff
@@ -463,5 +458,397 @@ private void ReadPropertyValueInODataErrorDetailObject(ODataErrorDetail detail,
+                }
+
+                this.AssertJsonCondition(JsonNodeType.Property, JsonNodeType.EndObject);
+                return typeName;
+            }
+
+            throw new ODataException(Strings.ODataJsonLightErrorDeserializer_PropertyAnnotationNotAllowedInErrorPayload(propertyAnnotationName));
+        }
+
+        /// <summary>
+        /// Asynchronously reads the JSON object which is the value of the "error" property.
+        /// </summary>
+        /// <param name="error">The <see cref="ODataError"/> object to update with data from the payload.</param>
+        /// <returns>A task that represents the asynchronous read operation.</returns>
+        /// <remarks>
+        /// Pre-Condition:  JsonNodeType.StartObject    - The start of the "error" object.
+        ///                 any                         - Will throw if not StartObject.
+        /// Post-Condition: any                         - The node after the "error" object's EndNode.
+        /// </remarks>
+        private Task ReadODataErrorObjectAsync(ODataError error)
+        {
+            return this.ReadJsonObjectInErrorPayloadAsync(
+                (propertyName, duplicationPropertyNameChecker) => this.ReadPropertyValueInODataErrorObjectAsync(error, propertyName, duplicationPropertyNameChecker));
+        }
+
+        /// <summary>
+        /// Asynchronously reads an inner error object.
+        /// </summary>
+        /// <param name="recursionDepth">The number of times this method has been called recursively.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains an <see cref="ODataInnerError"/> representing the read inner error.
+        /// </returns>
+        /// <remarks>
+        /// Pre-Condition:  JsonNodeType.StartObject    - The start of the "innererror" object.
+        ///                 any                         - will throw if not StartObject.
+        /// Post-Condition: any                         - The node after the "innererror" object's EndNode.
+        /// </remarks>
+        private async Task<ODataInnerError> ReadInnerErrorAsync(int recursionDepth)
+        {
+            Debug.Assert(this.JsonReader.DisableInStreamErrorDetection, "JsonReader.DisableInStreamErrorDetection");
+            this.JsonReader.AssertNotBuffering();
+
+            ValidationUtils.IncreaseAndValidateRecursionDepth(ref recursionDepth, this.MessageReaderSettings.MessageQuotas.MaxNestingDepth);
+
+            ODataInnerError innerError = new ODataInnerError();
+
+            await this.ReadJsonObjectInErrorPayloadAsync(
+                (propertyName, propertyAndAnnotationCollector) => this.ReadPropertyValueInInnerErrorAsync(recursionDepth, innerError, propertyName)).ConfigureAwait(false);
+
+            this.JsonReader.AssertNotBuffering();

```

Comment:
```suggestion
            this.JsonReader.AssertNotBuffering();

```

Reply:
Fixed

Reply:
What's the convention here? Should we add a new line after each statement? Or what criteria is being used to determined where to place the empty line?